types/V1.mli

```diff
@@ -102,6 +102,46 @@ module type CLOCK = sig
   (** Convert a time in seconds, as returned by {!time}, into a date
       and a time. Assumes UTC (Coordinated Universal Time), also known
       as GMT. *)
+
+end
+
+(** {1 Posix clock}
+
+    Clock returning time since the unix epoch. Subject to adjustment by eg. NTP.
+    Compatible with the Ptime library. *)
+module type PCLOCK = sig
+
+  val now_d_ps : unit -> int * int64
+  (** [now_d_ps ()] is [(d, ps)] representing the POSIX time occuring
+      at [d] * 86'400e12 + [ps] POSIX picoseconds from the epoch
+      1970-01-01 00:00:00 UTC. [ps] is in the range
+      \[[0];[86_399_999_999_999_999L]\]. *)
+
+  val current_tz_offset_s : unit -> int option
+  (** [current_tz_offset_s ()] is if available the clock's current local time zone
+      offset to UTC in seconds. *)
+
+  val period_d_ps : unit -> (int * int64) option
+  (** [period_d_ps ()] is if available [Some (d, ps)] representing the
+      clock's picosecond period [d] * 86'400e12 + [ps]. [ps] is in the
+      range \[[0];[86_399_999_999_999_999L]\]. *)
+
+end
+
+(** {1 Monotonic clock}
+
+    Clock returning monotonic time since an arbitrary point. To be used for eg.
+    profiling. *)
+
+module type MCLOCK = sig
+
+  val elapsed_ns : unit -> int64
+  (** [elapsed_ns ()] gives a monotonically increasing count in nanoseconds from
+   * some arbitrary point *)
+
+  val period_ns : unit -> int64 option
+  (** [period_ns ()] is if available [Some ns] representing the clock's
```

Comment:
I think I wrote this... but the sentence is difficult to parse. Commas are missing `is, if available,` or maybe the whole sentence is broken.